src/devices/cpu/mc68hc11/mc68hc11.cpp

```diff
@@ -384,6 +397,71 @@ void mc68hc11_cpu_device::tmsk2_w(uint8_t data)
 	m_tmsk2 = data;
 }
 
+void mc68hc11_cpu_device::serial_w(u8 val)
+{
+	m_scdr = val;
+	m_scsr |= 0x20;
+	if ((m_sccr2 & 0x20) != 0)
+		set_irq_state(0x14, true);
+}
+
+// unimplemented
+uint8_t mc68hc11_cpu_device::baud_r()
+{
+	return m_baud;
+}
+void mc68hc11_cpu_device::baud_w(uint8_t data)
+{
+	m_baud = data;
+}
+
+// unimplemented
+uint8_t mc68hc11_cpu_device::sccr1_r()
+{
+	return m_sccr1;
+}
+void mc68hc11_cpu_device::sccr1_w(uint8_t data)
+{
+	m_sccr1 = data;
+}
+
+uint8_t mc68hc11_cpu_device::sccr2_r()
+{
+	return m_sccr2;
+}
+void mc68hc11_cpu_device::sccr2_w(uint8_t data)
+{
+	m_sccr2 = data;
+
+	// TODO: check
+	if ((m_sccr2 & 0x80) != 0)
+		set_irq_state(0x14, true);
+}
+
+uint8_t mc68hc11_cpu_device::scsr_r()
+{
+	return m_scsr;
+}
+void mc68hc11_cpu_device::scsr_w(uint8_t data)
+{
+	m_scsr = data;
+}
+
+uint8_t mc68hc11_cpu_device::scdr_r()
+{
+	set_irq_state(0x14, false);
+	m_scsr |= 0b10000000;
```

Comment:
This needs to be protected against `!machine().side_effects_disabled()`, being a read access.